source/Ship.cpp

```diff
@@ -3374,7 +3423,41 @@ double Ship::MaxReverseVelocity() const
 // DamageDealt from that weapon. The return value is a ShipEvent type,
 // which may be a combination of PROVOKED, DISABLED, and DESTROYED.
 // Create any target effects as sparks.
-int Ship::TakeDamage(vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment)
+int Ship::TakeDamage(vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment, Point intersection, Point hitVelocity, Angle hitAngle)
+{
+	int type = DoDamage(visuals, damage);
+
+	// If this ship did not consider itself an enemy of the ship that hit it,
+	// it is now "provoked" against that government.
+	if(sourceGovernment && !sourceGovernment->IsEnemy(government)
+			&& (Shields() < .9 || Hull() < .9 || !personality.IsForbearing())
+			&& !personality.IsPacifist() && damage.GetWeapon().DoesDamage())
+		type |= ShipEvent::PROVOKE;
+
+	// Create hull/shield hit effect visuals at the point of impact for weapons
+	// fire, if there are any to create.
+	if(damage.Shield() && hasShieldHitEffect)
+	{
+		for(const auto &it : ShieldHitEffects())
```

Comment:
Don't call a getter function for data members that are part of this class.

Reply:
Replying for both this and your comment below: I'll remove the getter functions for these then and just directly grab the values instead.